src/main/java/uk/gov/cshr/civilservant/controller/CivilServantController.java

```diff
@@ -61,12 +82,44 @@ public CivilServantController(CivilServantRepository civilServantRepository,
         return getResourceResponseEntity(optionalCivilServant);
     }
 
+    @PatchMapping("/manager")
+    @PreAuthorize("isAuthenticated()")
+    public ResponseEntity<Resource<CivilServantResource>> check(@RequestParam(value = "email") String email) throws NotificationClientException {
+        IdentityFromService lineManager = new IdentityFromService();
+
+        Optional<CivilServant> optionalCivilServant = civilServantRepository.findByPrincipal();
+
+        if (optionalCivilServant.isPresent()) {
+            lineManager = lineManagerService.checkLineManager(email);
+
+
+            if (lineManager != null) {
+                CivilServant civilServant = optionalCivilServant.get();
+
+                civilServant = lineManagerService.UpdateAndNotifyLineManager(civilServant, lineManager, email);
```

Comment:
BUG - Sending notification even if civilServant is `null`. You have a null check on the next line!
SOLID - Self-mutating the variable. Why should line Manager Update and Nofify return a `CivllServant`?

Reply:
Civil servant is  gotten by findbyprinciple. It will always find the logged in user as there otherwise the user would not be able to update line manager. If there is a system error findbyprinciple will through an exception. That seems acceptable for /me.

LineManagerUpdateandNotify returns the updated civilservant and notifies the line manager of an update if needed. 

The point about SOLID is that it's open to interpretation, its not a deterministic process.

Reply:
Why would we notify if it is null?
Maybe your method need to be break into two methods?

Reply:
We do not notify  if  null .

optionalCivilServant.isPresent() checks that an optional civil servant is returned. That takes care if there is a civilservant is present so no null check is needed.   We do not notify if line manager does not exist.

UpdateAndNotifyLineManager returns an updated civil servant if the linemanager is not the same civilservant, in this case it returns null.The methed returns null if validation fails.

Validation of  data is appropriate for data storage and returning an entity on sucess or null on failure is appropriate. There is an argument for separating  out the notification part, however given that that code is a few lines, this is would be an exercise in semantics.